File: src/main/java/org/example/bankramenserver/infrastructure/mcp/auth/McpAgentPrincipal.java
package org.example.bankramenserver.infrastructure.mcp.auth;

import java.util.Set;
import java.util.UUID;

public record McpAgentPrincipal(
        String connectionId,
        String consumerId,
        UUID linkedUserId,
        Set<AgentScope> scopes
) {

    public boolean hasScope(AgentScope scope) {
        return scopes.contains(scope);
    }
}

File: src/main/kotlin/org/example/bankramenserver/infrastructure/mcp/auth/McpAgentPrincipal.kt
package org.example.bankramenserver.infrastructure.mcp.auth

import java.util.UUID

data class McpAgentPrincipal(
    val connectionId: String,
    val consumerId: String,
    val linkedUserId: UUID,
    val scopes: Set<AgentScope>,
) {
    fun hasScope(scope: AgentScope): Boolean = scope in scopes
}
